docs/overview.md

```diff
@@ -0,0 +1,385 @@
+
+Refs allow mutable "variables" in your program. They are a thin wrapper around
+a record with a mutable field called `contents`.
+
+Feature                         | Example
+--------------------------------|----------
+Type (already defined)          | `type ref('a) = {mutable contents: 'a};`
+Ref creation                    | `let x = ref(10);` or `let x = {contents: 10};`
+Ref access                      | `x^;` or `x.contents;`
+Ref update                      | `x := 20;` or `x.contents = 20;`
+
+## Loops
+
+_Details: [Loops](loops.md)_
+
+Loops are discouraged in most cases. Instead functional programming patterns
+like `map`, `filter`, or `reduce` can usually be used in their place.
+
+Feature                         | Example
+--------------------------------|----------
+While                           | `while (condition) {...}`
+For (incrementing)              | `for (i in 0 to 9) {...}` (inclusive)
+For (decrementing)              | `for (i in 9 downto 0) {...}` (inclusive)
+
+- Note: There is no `break` or early returns in Reason. Use a ref containing a
+bool for break-like behavior: `let break = ref(false); while (!break^ && condition) {...};`
+
+## Modules
+
+_Details: [Modules](modules)_
+
+Modules are a way to group types and values. Each Reason file implicitly
+creates a module of the same name. Each `type` definition and `let` binding in
+a module automatically becomes a "member" of that module which can be accessed
+by other modules.  Modules can also be nested using the `module` keyword.
+
+Feature                         | Example
+--------------------------------|----------
+Module creation                 | `module Foo = { let bar = 10; };`
+Module member access            | `Foo.bar;`
+Module types                    | `module type Foo = { let bar: int; };`
+
+## Functors
+
+Functors are like functions that create modules. This is an advanced topic
+that can be very powerful. Here is a basic example:
+
+```ocaml
+module type Stringable = sig
+  type t
+  val toString : t -> string
+end
+
+module Printer (Item : Stringable) = struct
+  let print (t : Item.t) = print_endline (Item.toString t)
+
+  let printList (list : Item.t list) =
+    list |> List.map Item.toString |> String.concat ", " |> print_endline
+end
+
+module IntPrinter = Printer (struct
+  type t = int
+  let toString = string_of_int
+end)
+
+let () =
+  IntPrinter.print 10; (* 10 *)
+  IntPrinter.printList [ 1; 2; 3 ] (* 1, 2, 3 *)
+```
+```reasonml
+module type Stringable = {
+  type t;
+  let toString: (t) => string;
+};
+
+module Printer = (Item: Stringable) => {
+  let print = (t: Item.t) => {
+    print_endline(Item.toString(t));
+  };
+
+  let printList = (list: list(Item.t)) => {
+    list
+    |> List.map(Item.toString)
+    |> String.concat(", ")
+    |> print_endline;
+  };
+};
+
+module IntPrinter = Printer({
+  type t = int;
+  let toString = string_of_int;
+});
+
+IntPrinter.print(10); // 10
+IntPrinter.printList([1, 2, 3]); // 1, 2, 3
```

Comment:
Maybe this part should be changed to 

```reason
let () = {
  IntPrinter.print(10); // 10
  IntPrinter.printList([1, 2, 3]); // 1, 2, 3
};
```
to better match the OCaml syntax version?